rasa/core/trackers.py

```diff
@@ -43,6 +43,17 @@ class EventVerbosity(Enum):
     ALL = 4
 
 
+class AnySlotDict(dict):
+    """A slot dictionary that pretends every slot exists, by creating slots on demand.
+
+    This only uses the generic slot type! This means certain functionality wont work,
+    e.g. properly featurizing the slot."""
+
+    def __missing__(self, key):
+        value = self[key] = Slot(key)
```

Comment:
I think it is better to use slot, as it will raise an exception if anyone tries to featurize it. Otherwise, this will lead to silent errors because the featurization won't be correct and since that will probably be used for some sort of prediction it will create wrong results.